lib/identity_cache/serializer.rb

```diff
@@ -0,0 +1,83 @@
+module IdentityCache
+  module Serializer # :nodoc:
+    DEHYDRATE_EVENT = "dehydration.identity_cache"
+    HYDRATE_EVENT   = "hydration.identity_cache"
+
+    class << self
+      def serialize(record)
+        return unless record
+
+        ActiveSupport::Notifications.instrument(DEHYDRATE_EVENT, class: record.class.name) do
+          coder_from_record(record, record.class)
+        end
+      end
+
+      def deserialize(coder, klass)
+        return unless coder
+
+        ActiveSupport::Notifications.instrument(HYDRATE_EVENT, class: klass.name) do
+          record_from_coder(coder, klass)
+        end
+      end
+
+      private
+
+      def coder_from_record(record, klass)
+        return unless record
+
+        {}.tap do |coder|
```

Comment:
Why not just use

```suggestion
        coder = {}
```

and add `coder` at the end of the method?  That would make the code simpler and reduce the indentation of these already long lines.

Reply:
Stylistic preferences I suppose. Still, I know using tap is sometimes polarizing so I've reverted to using regular assignment in both `coder_from_record` and `record_from_coder`.